tmt/steps/report/junit.py

```diff
@@ -266,7 +267,7 @@ class ReportJUnitData(tmt.steps.report.ReportStepData):
         default=DEFAULT_FLAVOR_NAME,
         option='--flavor',
         metavar='FLAVOR',
-        choices=[DEFAULT_FLAVOR_NAME, CUSTOM_FLAVOR_NAME],
+        choices=[DEFAULT_FLAVOR_NAME, CUSTOM_FLAVOR_NAME, 'polarion'],
```

Comment:
Hmm, can `junit` plugin really render the correct and meaningful `polarion` flavor? IIUIC, it cannot, because there is plenty of code in the `polarion` plugin precrunching information for the template. So, `--how junit --flavor polarion` is probably not going to work as one might expect.

In this sense, `polarion` is probably not a flavor owned by the `junit` plugin, and it can and should be isolated from the `junit` plugin as any other 3rd party junit-rendering plugin there might be. Both are shipped with tmt, sure, but on the implementation level, `polarion` brings this "flavor", by providing a template and calling `junit` functionality, all that while `junit` is not aware of `polarion` existence. At least not by its name :)

There is also `polarion` being mentioned by name in the template below, would it be possible to make that point generic enough for `polarion` template to take over control and render `properties`? What I'm worried about: let's say a few more flavors appear out there, in the wild world, `pytest` and `jest`, and these would also like to include `<properties/>` elements. Will we add their names to the `{% if (FLAVOR == 'polarion') and result.properties is defined %}` condition as well? I suspect we wouldn't, which makes me think this "leaking" of `polarion` into `_base.xml.j2` is not right - it's necessary to let `polarion` emit this element, but can we do it in way in which `_base.xml.j2` is not guarding it with a name-based check?

Reply:
As discussed yesterday, I have moved the properties decorator into the `junit.py` module. I also removed the `flavor` condition for polarion from the `_base.xml.j2` template and the `polarion` from the supported arguments of the `--flavor` argument. The `polarion` flavor is now used only internally by polarion report plugin.